File: ruby/hamming/hamming.rb
class Hamming
  def self.compute(original, mutation)
    # check equal length
    unless original.length == mutation.length
      raise ArgumentError
    end

    # go through both strings
    difference = 0
    mutation_char = mutation.split("")
    original.each_char.with_index(0) do |character, index|
      unless mutation_char[index] == character.to_s
        difference = difference + 1
      end
    end

    difference
  end
end

File: javascript/hamming/hamming.js
var Hamming = function() {};

Hamming.prototype.compute = function(original, mutation) {
  if (original.length !== mutation.length) {
    throw 'DNA strands must be of equal length.';
  }

  var count = 0;
  for (var i = 0; i < original.length; i++) {
    if (original[i] !== mutation[i]) {
      count++;
    }
  }
  return count;
}

module.exports = Hamming;